presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java

```diff
@@ -293,11 +293,12 @@ private int readAllNulls(int[] positions, int positionCount)
             throws IOException
     {
         presentStream.skip(positions[positionCount - 1]);
-
+        outputPositionCount = 0;
```

Comment:
This statement belongs to just before the for loop `for (int i = 0; i < positionCount; i++) {` (outputPositionCount get reassigned immediately in the else branches that follow).